go/vt/vtgate/planbuilder/testdata/aggr_cases.txt

```diff
@@ -5774,3 +5774,42 @@ Gen4 plan same as above
     "user.user"
   ]
 }
+
+# Group By X Order By X
+"SELECT user.intcol FROM user GROUP BY user.intcol ORDER BY COUNT(user.intcol)"
+"unsupported: in scatter query: complex order by expression: count(`user`.intcol)"
+{
+  "QueryType": "SELECT",
+  "Original": "SELECT user.intcol FROM user GROUP BY user.intcol ORDER BY COUNT(user.intcol)",
+  "Instructions": {
+    "OperatorType": "Sort",
+    "Variant": "Memory",
+    "OrderBy": "1 ASC",
+    "ResultColumns": 1,
+    "Inputs": [
+      {
+        "OperatorType": "Aggregate",
+        "Variant": "Ordered",
+        "Aggregates": "sum_count(1) AS count(`user`.intcol)",
+        "GroupBy": "0",
+        "Inputs": [
+          {
+            "OperatorType": "Route",
+            "Variant": "Scatter",
+            "Keyspace": {
+              "Name": "user",
+              "Sharded": true
+            },
+            "FieldQuery": "select `user`.intcol, count(`user`.intcol) from `user` where 1 != 1 group by `user`.intcol",
+            "OrderBy": "0 ASC",
+            "Query": "select `user`.intcol, count(`user`.intcol) from `user` group by `user`.intcol order by `user`.intcol asc",
```

Comment:
Shouldn't we be able to make this `ORDER BY NULL` because we don't care about the order of results here? 🤔

Reply:
Or, alternatively, the `ORDER BY` part could be removed completely. But IIRC, specifying `ORDER BY NULL` will perform better in some cases.

Reply:
@arthurschreiber, we do want to `ORDER BY` the columns on which we do `GROUP BY` so they're properly ordered when doing the grouping.